authors/apps/search/views.py

```diff
@@ -0,0 +1,60 @@
+from django_filters import rest_framework as filters
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework import generics
+from rest_framework.filters import SearchFilter
+from rest_framework.permissions import AllowAny
+
+from authors.apps.articles.models import Articles, Favorite
+from authors.apps.articles.renderers import ArticleJSONRenderer
+from authors.apps.articles.serializers import ArticleSerializer
+from authors.apps.authentication.models import User
+
+
+class ArticleFilter(filters.FilterSet):
+    author = filters.CharFilter(field_name='author__username', lookup_expr='exact')
+    title = filters.CharFilter(field_name='title', lookup_expr='icontains')
+
+    class Meta:
+        model = Articles
+        fields = ['author', 'title', 'created_at']
+
+
+class SearchArticleListAPIView(generics.ListAPIView):
+    permission_classes = (AllowAny,)
+    serializer_class = ArticleSerializer
+    queryset = Articles.objects.all()
+    renderer_classes = (ArticleJSONRenderer,)
+
+    filter_backends = (DjangoFilterBackend, SearchFilter)
+    filterset_class = ArticleFilter
+    search_fields = ('author__username', 'title', 'created_at',)
+
+    def get_queryset(self):
+        # Getting specific properties from the request body
+        favorited = self.request.query_params.get('favorited', '')  # returns username
+        tags = self.request.query_params.get('tags', '')  # returns tags
+
+        if favorited:
+            user = User.objects.get(username=favorited)
+            # list of favourites query set by the user 
+            favorite_queryset = Favorite.objects.filter(user_id=user)
+            # This loops over all objects inside the favorite_queryset variable
+            # and returns a list of all article id's
+            article_ids = [fav.article_id.id for fav in favorite_queryset]
```

Comment:
Suggestion: You could look into using `values_list` to return a list of `article_ids` when querying so you don't have to create a new list comprehension

Reply:
I have implemented it. Kindly review it.